### ISSUE TITLE
[data grid] The provided aria label prop is attached on the wrong grid element, causing issues with accessibility

### ISSUE THREAD

**Issue by lemmylem** [mui/mui-x#13521]
### The problem in depth

I am a MUI Pro license user and customers are experiencing accessibility issues with the data grid. The problem is that the aria label that MUI provides is attached to the "wrong" element. It should be attached to the div with the `role="grid"`, so screen readers can pick it up.

Currently, for example, the NVDA screen reader does not correctly read the aria label, in the grid, failing our accessibility requirements. In the below image, the red circle is where the aria label is currently attached, and the green is where it should be (or vise versa). The overall requirement is that the role must be on the same div as the aria label, or else there will be issues with screen readers reading the correct back to the user.

<img width="1323" alt="aria-label" src="https://github.com/mui/mui-x/assets/139143259/6df5d4a7-a844-46de-b554-5e9b93d320dd">

### Your environment

**OrderID:** 8813014139

**Search keywords**: aria, label

**Comment by timokasa** [comment]
This is very much still an issue.

Originally the aria-label and aria-labelledby props were implemented as a solution for this issue: https://github.com/mui/mui-x/issues/1700
Before Datagrid v7 the role="grid" was set on the MuiDataGrid-root Element and passing aria-label or aria-labelledby was also set on the MuiDataGrid-root  Element. The Screenreader picked it up and it worked as expected.

In v7 the role="grid" is no longer set on MuiDataGrid-root Element but on MuiDataGrid-main Element. But the aria-label and aria-labelledby are set on the Parent MuiDataGrid-root Element. This prevents the Screenreader from picking it up.

The issue in the v8 beta is still persistent.

I was not able to found a way to set the aria-label or aria-labelledby on the MuiDataGrid-main Element, as slotProps.main are not supported.

I can see 2 ways to fix this issue:
1) extending useGridAriaAttributes in GridMainContainer.tsx to include the aria-label and aria-labelledby from the root props
2) enabling slotProps.main so we can set the aria props directly

First option would be better i think.

Please fix this!

Support Key: 106776
